xsd-core/src/main/java/io/ballerina/xsdtorecordconverter/XSDFactory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.xsdtorecordconverter;
+
+import io.ballerina.xsdtorecordconverter.component.ComplexType;
+import io.ballerina.xsdtorecordconverter.component.Element;
+import io.ballerina.xsdtorecordconverter.component.XSDComponent;
+import io.ballerina.xsdtorecordconverter.component.SimpleType;
+import org.w3c.dom.Node;
+
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.COMPLEX_TYPE;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.ELEMENT;
+import static io.ballerina.xsdtorecordconverter.visitor.XSDVisitorImpl.SIMPLE_TYPE;
+
+/**
+ * The factory class for generating XSD components.
+ */
+public final class XSDFactory {
+    public static XSDComponent generateComponents(Node node) {
```

Comment:
Can't we change the method signature like this ?

```java
public static Optional<XSDComponent> generateComponents(Node node)
```

I think this is a more cleaner approach rather than using the `null`.